src/main/java/rx/internal/util/ScalarSynchronousObservable.java

```diff
@@ -29,28 +34,105 @@
     private final T t;
 
     protected ScalarSynchronousObservable(final T t) {
-        super(new OnSubscribe<T>() {
+        super(new ScalarSynchronousObservableOnSubscribe<T>(t));
+        this.t = t;
+    }
+    private static final class ScalarSynchronousObservableOnSubscribe<T> implements OnSubscribe<T> {
+        private final T t;
 
-            @Override
-            public void call(Subscriber<? super T> s) {
-                /*
-                 *  We don't check isUnsubscribed as it is a significant performance impact in the fast-path use cases.
-                 *  See PerfBaseline tests and https://github.com/ReactiveX/RxJava/issues/1383 for more information.
-                 *  The assumption here is that when asking for a single item we should emit it and not concern ourselves with 
-                 *  being unsubscribed already. If the Subscriber unsubscribes at 0, they shouldn't have subscribed, or it will 
-                 *  filter it out (such as take(0)). This prevents us from paying the price on every subscription. 
-                 */
-                s.onNext(t);
+        private ScalarSynchronousObservableOnSubscribe(T t) {
+            this.t = t;
+        }
+
+        @Override
+        public void call(Subscriber<? super T> s) {
+            /*
+             *  We don't check isUnsubscribed as it is a significant performance impact in the fast-path use cases.
+             *  See PerfBaseline tests and https://github.com/ReactiveX/RxJava/issues/1383 for more information.
+             *  The assumption here is that when asking for a single item we should emit it and not concern ourselves with 
+             *  being unsubscribed already. If the Subscriber unsubscribes at 0, they shouldn't have subscribed, or it will 
+             *  filter it out (such as take(0)). This prevents us from paying the price on every subscription. 
+             */
+            s.onNext(t);
+            // checking unsubscribe here because if they really wanted performance then they should have called get().
+            if (!s.isUnsubscribed())
                 s.onCompleted();
+        }
+    }
+
+    public T get() {
+        if (onSubscribe instanceof ScalarSynchronousObservableOnSubscribe)
+            return t;
+
+        // Slow path because the observable execution hooks are in effect.  We
+        // have no idea what the onSubscribe is going to do anymore so this code
+        // is being cautious and doesn't assume that it is going to scalar or
+        // synchronous anymore.  I intentionally also didn't use toBlocking()
+        // or any other operators to avoid adding noise to whatever the hooks
+        // are trying to record.  The subscribe below simulates the full
+        // subscribe/onNext/onComplete/unsubscribe life cycle of this observable.
+
+        // use a latch just in case it is asynchronous
+        final CountDownLatch done = new CountDownLatch(1);
```

Comment:
`ScalarSynchronousObservable.get()` is already a blocking operation. The instance of check at the top short circuits the complicated logic below.